front/components/assistant_builder/ActionsScreen.tsx

```diff
@@ -182,26 +182,34 @@ export default function ActionsScreen({
           }
 
           // Making sure the name is not used already.
+          let newActionName = newAction.name;
           let index = 1;
-          const newNameWithoutSuffix = newAction.name.replace(/_\d+$/, "");
-          const suffixedName = () =>
-            index > 1
-              ? `${newNameWithoutSuffix}_${index}`
-              : newNameWithoutSuffix;
-          while (builderState.actions.some((a) => a.name === suffixedName())) {
+          let isNameUsed = builderState.actions.some(
+            (a) => a.name === newActionName
+          );
+          while (isNameUsed) {
+            newActionName = `${newAction.name.replace(/_\d+$/, "")}_${index}`;
```

Comment:
```suggestion
            newActionName = `${newAction.name.replace(/_\d+$/, "_${index}")}`;
```

Reply:
I tried this and it breaks, so keeping as is for now but added a todo for monday to improve this code

newActionName = `${newAction.name.replace(/_\d+$/, `_${index}`)}`;